src/Datadog.Trace/Agent/ExporterWriterBuffer.cs

```diff
@@ -20,19 +20,14 @@ public bool Push(T item)
         {
             lock (_items)
             {
-                if (_count < _items.Length)
+                if (_count >= _items.Length)
                 {
-                    _items[_count++] = item;
-                    return true;
-                }
-                else
-                {
-                    // drop a random trace.
-                    // note that Random.Next() is NOT thread-safe,
-                    // but we are running this inside a lock
-                    _items[_random.Next(_items.Length)] = item;
+                    // drop the trace as the buffer is full
```

Comment:
This looks like it will effectively change the sampling strategy (under certain situations), and as a result we may want to consider making a change like this in a separate PR. At some point we'll also want to update the sampling strategy to match the strategies that OTel requires, so we may want to delay a change like this until then.

Reply:
We can remove this change per https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/74#issuecomment-794679841

Reply:
Moved to https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/81 where we can have a dedicated discussion.